### PR TITLE
[#7531,#7532] Bump version number and roll our own CMake package version file (4-3-stable)

### PR DESCRIPTION
**Pull request by SwooshyCueb**
In service of #7531
Addresses #7532

Submitting this separately from my libstdc++ PRs, as I want it to land first.

Companion PRs:
- irods/irods_rule_engine_plugin_audit_amqp#136
- irods/irods_client_icommands#445
- irods/irods_capability_storage_tiering#263
- irods/irods_microservice_plugins_curl#66
- irods/irods_netcdf#54
- irods/irods_resource_plugin_s3#2176
- irods/irods_rule_engine_plugin_logical_quotas#108
- irods/irods_rule_engine_plugin_metadata_guard#77
- irods/irods_rule_engine_plugin_python#195

### CODE REVIEW

**Comment by SwooshyCueb** [comment]
This PR and the listed companion PRs are ready for review

File: cmake/IRODSConfigVersion.cmake.in [hunk 1]
```diff
@@ -50,12 +50,24 @@ endif()
...
+  # inclusive ("INCLUDE") or exclusive ("EXCLUDE"). (At time of writing, CMake only supports
+  # "INCLUDE" for the lower end, but the way the documentation is written implies that "EXCLUDE"
+  # might someday be supported, and supporting that possibility here is trivial, so we do.)
+  # PACKAGE_FIND_VERSION_MAX is the upper end of the version range, and
+  # PACKAGE_FIND_VERSION_RANGE_MAX tells us whether the lower end of the version range is
+  # inclusive ("INCLUDE") or exclusive ("EXCLUDE").
   if (
+    # lower end of version range is inclusive and iRODS version is older than requested range
     ((PACKAGE_FIND_VERSION_RANGE_MIN STREQUAL "INCLUDE" AND PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION_MIN)
+      # lower end of version range is exclusive and iRODS version is older than or euqal to requested range
```
Comment (korydraughn) [inline cmake/IRODSConfigVersion.cmake.in:66]: euqal -> equal

Comment (SwooshyCueb) [reply cmake/IRODSConfigVersion.cmake.in]: fixed

**Review by korydraughn — COMMENTED** [review]
This looks very similar to the PR for the main branch.

Are there any differences?

**Comment by SwooshyCueb** [comment]
> This looks very similar to the PR for the main branch.
> 
> Are there any differences other than the version bump to 4.3.2?

Nope!

**Review by korydraughn — APPROVED** [review]

**Review by alanking — APPROVED** [review]
\# it when ready

**Comment by SwooshyCueb** [comment]
\#'d

**Pull request merged**